src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -658,6 +658,31 @@ public ProjectState UpdateAdditionalDocument(TextDocumentState newDocument, bool
                 latestDocumentTopLevelChangeVersion: dependentSemanticVersion);
         }
 
+        public ProjectState UpdateDocumentsOrder(ImmutableArray<DocumentId> documentIds)
+        {
+            if (documentIds.Length != _documentIds.Count)
+            {
+                throw new ArgumentException("The count of the specified document ids do not match the count of the document ids in the project.", nameof(documentIds));
+            }
+
+            var orderedDocumentIds = ImmutableList.CreateBuilder<DocumentId>();
+
+            foreach(var documentId in documentIds)
+            {
+                if (this.ContainsDocument(documentId))
+                {
+                    orderedDocumentIds.Add(documentId);
```

Comment:
you don't really need this separate list, since it will just end up exactly the same as documentIds

Reply:
Good catch.